src/utils.cpp

```diff
@@ -739,4 +740,31 @@ std::string hex_format_buffer(const char *buf, size_t size)
   return std::string(s);
 }
 
+std::unordered_set<std::string> get_traceable_funcs()
+{
+  // Try to get the list of functions from BPFTRACE_AVAILABLE_FUNCTIONS_TEST env
+  const char *path = std::getenv("BPFTRACE_AVAILABLE_FUNCTIONS_TEST");
+
+  // Use kprobe list as default
+  if (!path)
+    path = kprobe_path.c_str();
+
+  std::ifstream available_funs(path);
+  if (available_funs.fail())
+  {
+    if (bt_debug != DebugLevel::kNone)
+    {
+      std::cerr << "Error while reading traceable functions from "
+                << kprobe_path << ": " << strerror(errno);
```

Comment:
Hmm, this seems unlikely to fail. But when it does fail, kfuncs are disabled. I think this message would help user figure out what went wrong in that case. Not sure if it should be hidden behind `-d` flag.

Reply:
That's true, although I don't think that the message should be shown every time the user lists probes and the file is not available. I'd suggest leaving it behind `-d` flag and displaying a similar message when user tries to attach a kfunc and it fails due to this error. I'd implement that as a separate PR that will improve error messages for kfuncs in general.

Reply:
Why not raise an exception? That way the caller can figure out what to do with it which makes more sense for a "library" function like this imo.

If kfuncs depend on this we should probably hook it into `--info` too, would be confusing if bpftrace reports that kfuncs should work when it doesn.t

Reply:
> Why not raise an exception? That way the caller can figure out what to do with it which makes more sense for a "library" function like this imo.

I agree that raising an exception is cleaner, however, the caller is always the same - the constructor of `BTF`, called from the constructor of `BPFtrace`, so we would end up with the same problem again. A solution could be to postpone reading of the file to the phase when the function list is actually queried, but the function doing it (`BTF::get_funcs`) is marked as `const` and I didn't want to break that.

> If kfuncs depend on this we should probably hook it into `--info` too, would be confusing if bpftrace reports that kfuncs should work when it doesn.t

I agree, will do.